.github/workflows/sycl_detect_changes.yml

```diff
@@ -47,4 +47,4 @@ jobs:
               - *xptifw
               - *libclc
               - 'sycl/*'
-              - 'sycl/!(test-e2e)/**'
+              - 'sycl/!(test-e2e|doc)/**'
```

Comment:
What does "Don't treat sycl/doc/** as a functional change" mean?
This treats changes in `test-e2e` and `doc` directories the same way, but if there changes in `test-e2e`, we need to run these tests and to not run them if only documentation changes. I think this change says that "do not run SYCL compile-only tests if only documentation has changed". Do we still run test-e2e tests? If so I think we still "treat sycl/doc/** as a functional change" and PR title should be adjusted.